client/components/tinymce/plugins/wpcom/plugin.js

```diff
@@ -289,10 +289,15 @@ function wpcomPlugin( editor ) {
 
 	// Remove spaces from empty paragraphs.
 	editor.on( 'BeforeSetContent', function( event ) {
-		var paragraph = tinymce.Env.webkit ? '<p><br /></p>' : '<p></p>';
-
 		if ( event.content ) {
-			event.content = event.content.replace( /<p>(?:&nbsp;|\u00a0|\uFEFF|\s)+<\/p>/gi, paragraph );
+			var paragraph = tinymce.Env.webkit ? '<p><br /></p>' : '<p></p>';
+
+			event.content = event.content.replace( /<p>([^<>]+)<\/p>/gi, function( tag, text ) {
```

Comment:
I'm trying to understand what is the goal of this `RegExp`. Here is what I am guessing:

 - Only match `<p>` tags with no attributes such as `class` or `style` or anything else.
 - Find any content between these `<p>` tags excluding the angle brackets.
   - If that content is nothing but whitespace characters, replace the inner content with the predefined empty paragraph markup.

Further question, is the goal to transform `<p>&nbsp;</p>` into `<p><p></p></p>`?